x/ecocredit/client/marketplace/proposal.go

```diff
@@ -0,0 +1,85 @@
+package marketplace
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govclient "github.com/cosmos/cosmos-sdk/x/gov/client"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/rest"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/spf13/cobra"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+var AllowedAskDenomProposal = govclient.NewProposalHandler(TxAllowedAskDenomProposal, func(context client.Context) rest.ProposalRESTHandler {
+	return rest.ProposalRESTHandler{
+		SubRoute: "",
+		Handler:  nil,
+	}
+})
+
+func TxAllowedAskDenomProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "ask-denom-proposal [path_to_file.json] [flags]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a proposal for a new allowed ask denom",
+		Long: strings.TrimSpace(`Submit a proposal to add a new allowed ask denom. 
+The json file MUST take the following form:
+{
+	"title": "some title",
+	"description": "some description",
+	"allowed_denom": {
+					"bank_denom": "uregen",
+					"display_denom": "regen",
+					"exponent": 18
+	}
+}
+The bank denom is the underlying coin denom (i.e. ibc/CDC4587874B85BEA4FCEC3CEA5A1195139799A1FEE711A07D972537E18FD). 
+Display denom is used for display purposes, and serves as the name of the coin denom (i.e. ATOM). Exponent is the 
+amount micro coins makes up one whole coin. TODO(Tyler): not sure about this description...`),
```

Comment:
wasn't quite sure how to describe the exponent... any suggestions?

Reply:
Take a look at the basket description for exponent in `MsgCreate` but rather than credits we are converting between the denom and the display denom. Also see `exponent` in `AllowedDenom`.